core/templates/domain/question/QuestionObjectFactory.ts

```diff
@@ -17,149 +17,178 @@
  * question domain objects.
  */
 
-require('domain/state/StateObjectFactory.ts');
-
-angular.module('oppia').factory('QuestionObjectFactory', [
-  'StateObjectFactory', 'DEFAULT_LANGUAGE_CODE', 'INTERACTION_SPECS',
-  function(StateObjectFactory, DEFAULT_LANGUAGE_CODE, INTERACTION_SPECS) {
-    var Question = function(id, stateData, languageCode, version,
-        linkedSkillIds) {
-      this._id = id;
-      this._stateData = stateData;
-      this._languageCode = languageCode;
-      this._version = version;
-      this._linkedSkillIds = linkedSkillIds;
-    };
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import cloneDeep from 'lodash/cloneDeep';
+
+import { InteractionSpecsConstants } from 'pages/interaction-specs.constants';
+import { StateObjectFactory, State } from 'domain/state/StateObjectFactory';
+
+export interface IQuestionBackendDict {
+  id: string;
+  // eslint-disable-next-line camelcase
+  question_state_data: State;
+  // eslint-disable-next-line camelcase
+  language_code: string;
+  version: number;
+  // eslint-disable-next-line camelcase
+  linked_skill_ids: string[];
+}
+
+export class Question {
+  id: string;
```

Comment:
Shouldn't we use the `private` keyword for this instead of underscore as @brianrodri mentioned?

Reply:
> In the skill editor, when the questions tab is opened, I get this error:
> ![image](https://user-images.githubusercontent.com/22347970/80287027-b4df0380-874c-11ea-9e04-821efc0cc735.png)
> 
> Though, it works fine in topic editor. You might need to replicate the import in topic-editor-page.module in skill editor as well.

@aks681 If that's the case for skill editor then it might be the case also for the rest of the modules such as `classroom-page.module.ts` and more?